### PR TITLE
Count Julia docstrings separately from code

### PR DESCRIPTION
**Pull request by ericphanson**
Closes https://github.com/JuliaEcosystem/PackageAnalyzer.jl/issues/76

This PR uses JuliaSyntax to count Julia code. Split out from #79 since I think this part is better scoped and easier to land by itself.

This should be non-breaking from an API and an serialization perspective, although obviously the counts change.

TODO:
- [x] Tests for the code-counting functionality

### CODE REVIEW

File: src/PackageAnalyzer.jl [hunk 1]
```diff
@@ -20,7 +23,36 @@ using RegistryInstances: RegistryInstances, reachable_registries, PkgEntry
...
+##
+# Borrowed from
+# https://github.com/beacon-biosignals/SlackThreads.jl/blob/74351c2863ec9a1cf22732873d4d2816aa9c140d/src/SlackThreads.jl#L27-L49
+const CATCH_EXCEPTIONS = Ref(true)
+
+# We turn off exception handling for our tests, to ensure we aren't throwing exceptions
+# that we're missing. But we have it on by default, since in ordinary usage we want to
+# be sure we are catching all exceptions.
+# We wrap all public API methods in this, which should make it very difficult to throw
+# an exception.
```
Comment (ericphanson) [inline src/PackageAnalyzer.jl:37]:
```suggestion
```

**Review by giordano — APPROVED** [review]
I only had a cursory look, also because I'm not much familiar with JuliaSyntax, but this looks awesome!

Should we have some tests to ensure we can still read V1 of the schema?

**Comment by ericphanson** [comment]
https://github.com/JuliaEcosystem/PackageAnalyzer.jl/pull/86/files#diff-3b9314a6f9f2d7eec1d0ef69fa76cfabafdbe6d0df923768f9ec32f27a249c63R407 should cover that, since it was serialized out w/ V1 of the schema (and the file isn't updated in this PR).

BTW: should this be a breaking or feature release for PackageAnalyzer? It doesn't break serialization but the actual counts do change, since now we count docstrings.

**Comment by giordano** [comment]
> [#86 (files)](https://github.com/JuliaEcosystem/PackageAnalyzer.jl/pull/86/files#diff-3b9314a6f9f2d7eec1d0ef69fa76cfabafdbe6d0df923768f9ec32f27a249c63R407) should cover that, since it was serialized out w/ V1 of the schema (and the file isn't updated in this PR).

Uh, ok!

> BTW: should this be a breaking or feature release for PackageAnalyzer? It doesn't break serialization but the actual counts do change, since now we count docstrings.

Even though I don't think this is technically breaking, keeping consistent numbers within a major release is probably a safe choice!

**Pull request merged**